### PR TITLE
App: Ability to select/deselect all fields on export

### PR DESCRIPTION
**Pull request by joselcvarela**
## Description
If you have many fields on a collection, it can be tedious to select them all, if that is what you want.

This provides the ability to select/deselect all fields with a single click.

### Before

https://user-images.githubusercontent.com/14039341/193322475-45b19ad1-6995-423d-a7a3-39a83043f806.mov

### After

https://user-images.githubusercontent.com/14039341/193322500-fb35baee-44b8-4a99-85cd-35bc131b0d1e.mov

Closes ENG-16

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [x] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: app/src/components/v-field-list/v-field-list.vue [hunk 1]
```diff
@@ -37,16 +48,18 @@ interface Props {
 	disabledFields?: string[];
 	includeFunctions?: boolean;
 	includeRelations?: boolean;
+	canSelectAll?: boolean;
```
Comment (rijkvanzanten) [inline app/src/components/v-field-list/v-field-list.vue:51]: Lets rename this to `allowSelectAll` to be consistent with the `v-select`

File: app/src/views/private/components/export-sidebar-detail.vue [hunk 2]
```diff
@@ -211,6 +211,7 @@
 					<interface-system-fields
 						:value="exportSettings.fields"
 						:collection-name="collection"
+						:can-select-all="true"
```
Comment (rijkvanzanten) [inline app/src/views/private/components/export-sidebar-detail.vue:214]:
```suggestion
						can-select-all
```

(and rename to `allow-select-all` as well ;) )

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/views/private/components/export-sidebar-detail.vue (hunk 2 above)
Comment (joselcvarela) [reply app/src/views/private/components/export-sidebar-detail.vue]: Fixed 👍

File: app/src/components/v-field-list/v-field-list.vue (hunk 1 above)
Comment (joselcvarela) [reply app/src/components/v-field-list/v-field-list.vue]: Fixed 👍

**Comment by Shubhangi-ey** [comment]
With this feature will we able to import/export all the children, sub children data ?

**Comment by joselcvarela** [comment]
> With this feature will we able to import/export all the children, sub children data ?

Yes, it should. As you can see in "After" video, the nested fields "Title", "Description", etc. are being selected inside field "Filte"

**Comment by Shubhangi-ey** [comment]
> ested fields "Title", "Desc

It means it will also support more than one level of nested children. Any limit to the level of nesting ?

**Review by br41nslug — APPROVED** [review]
LGTM

**Comment by licitdev** [comment]
Is https://github.com/directus/directus/pull/17122 a duplicate of this PR? 🤔

**Comment by br41nslug** [comment]
Looks like it 🤔

**Comment by freekrai** [comment]
we are going to close this PR in favour of #17122 which is a simpler approach to adding select all / deselect all.

**Pull request closed without merging**